primitives/src/merkle_tree/universal_merkle_tree.rs

```diff
@@ -146,6 +146,50 @@ where
     }
 }
 
+impl<E, H, I, Arity, T> PersistentUniversalMerkleTreeScheme
+    for UniversalMerkleTree<E, H, I, Arity, T>
+where
+    E: Element,
+    H: DigestAlgorithm<E, I, T>,
+    I: Index + ToTraversalPath<Arity>,
+    Arity: Unsigned,
+    T: NodeValue,
+{
+    fn persistent_update(
+        &self,
+        pos: impl Borrow<Self::Index>,
+        elem: impl Borrow<Self::Element>,
+    ) -> Result<Self, PrimitivesError> {
+        self.persistent_update_with(pos, |_| Some(elem.borrow().clone()))
+    }
+
+    fn persistent_remove(&self, pos: Self::Index) -> Result<Self, PrimitivesError> {
+        self.persistent_update_with(pos, |_| None)
+    }
+
```

Comment:
these can be default impl?

Reply:
Yes, also for `update` and `remove` [6946a32](https://github.com/EspressoSystems/jellyfish/pull/489/commits/6946a320d112ac69e289cc1525bedafaa1219564)